sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/IcebergCatalogConfig.java

```diff
@@ -19,212 +19,39 @@
 
 import com.google.auto.value.AutoValue;
 import java.io.Serializable;
+import java.util.Properties;
 import javax.annotation.Nullable;
-import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.collect.ImmutableMap;
+import org.apache.beam.sdk.util.ReleaseInfo;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.collect.Maps;
 import org.apache.hadoop.conf.Configuration;
-import org.apache.iceberg.CatalogProperties;
 import org.apache.iceberg.CatalogUtil;
 import org.checkerframework.dataflow.qual.Pure;
 
 @AutoValue
 public abstract class IcebergCatalogConfig implements Serializable {
-
-  @Pure
-  public abstract String getName();
-
-  /* Core Properties */
-  @Pure
-  public abstract @Nullable String getIcebergCatalogType();
-
-  @Pure
-  public abstract @Nullable String getCatalogImplementation();
-
-  @Pure
-  public abstract @Nullable String getFileIOImplementation();
-
-  @Pure
-  public abstract @Nullable String getWarehouseLocation();
-
-  @Pure
-  public abstract @Nullable String getMetricsReporterImplementation();
-
-  /* Caching */
-  @Pure
-  public abstract boolean getCacheEnabled();
-
-  @Pure
-  public abstract boolean getCacheCaseSensitive();
-
-  @Pure
-  public abstract long getCacheExpirationIntervalMillis();
-
-  @Pure
-  public abstract boolean getIOManifestCacheEnabled();
-
-  @Pure
-  public abstract long getIOManifestCacheExpirationIntervalMillis();
-
-  @Pure
-  public abstract long getIOManifestCacheMaxTotalBytes();
-
-  @Pure
-  public abstract long getIOManifestCacheMaxContentLength();
-
-  @Pure
-  public abstract @Nullable String getUri();
-
-  @Pure
-  public abstract int getClientPoolSize();
-
-  @Pure
-  public abstract long getClientPoolEvictionIntervalMs();
-
-  @Pure
-  public abstract @Nullable String getClientPoolCacheKeys();
-
-  @Pure
-  public abstract @Nullable String getLockImplementation();
-
-  @Pure
-  public abstract long getLockHeartbeatIntervalMillis();
-
-  @Pure
-  public abstract long getLockHeartbeatTimeoutMillis();
-
-  @Pure
-  public abstract int getLockHeartbeatThreads();
-
-  @Pure
-  public abstract long getLockAcquireIntervalMillis();
-
-  @Pure
-  public abstract long getLockAcquireTimeoutMillis();
-
-  @Pure
-  public abstract @Nullable String getAppIdentifier();
-
-  @Pure
-  public abstract @Nullable String getUser();
-
   @Pure
-  public abstract long getAuthSessionTimeoutMillis();
+  public abstract Properties getProperties();
```

Comment:
@kennknowles any benefits to using `java.util.Properties` as opposed to a `Map<String, String>`?